.github/workflows/TEST_unit.yml

```diff
@@ -0,0 +1,35 @@
+name: Unit Tests
+
+on:
+  push:
+    branches: '*'
+  pull_request:
+    branches: '*'
+
+jobs:
+  test:
+    timeout-minutes: 5
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node-version: [20]
+
+    steps:
+      - name: 🛎️ Checkout Code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0 # All history for branches and tags
+
+      - name: 🍞 Install Bun
+        uses: oven-sh/setup-bun@v2
+        with:
+          bun-version: latest
+
+      - name: 📂 Install Dependencies
+        run: bun install
+
+      - name: 💀 Run Unit Tests
+        run: bun run test:vitest
```

Comment:
make sure we rename this accordingly as well

Reply:
Doing some important work and can't stash, feel free to change this and force push

Reply:
Also, you want you keep the commands generic, makes mass migrations much easier. So the script for `test:unit` could use Jest in the future if necessary.